README.md

```diff
@@ -187,3 +187,101 @@ After that you can start QEMU
 ```shell
 qemu-system-x86_64 -serial stdio -drive file=trenchboot.img,if=virtio -enable-kvm
 ```
+
+## Development
+
+Usually after making changes to recipes it's enough to just run [build](#build)
+again but sometimes not all changes are detected. In those cases you can clean
+changed recipe and then rebuild the project.
+
+```shell
+kas-container shell meta-trenchboot/kas-generic-tb.yml -c "bitbake -c cleansstate <recipes>"
+kas-container build meta-trenchboot/kas-generic-tb.yml
+```
+
+Replace `<recipes>` with name of changed recipes e.g. `skl`, `grub`,
+`linux-tb` and then rebuild.
+
+### Source revision
+
+To change branch or commit used by a recipe you have to change `BRANCH` or
+`SRCREV` variable in appropriate recipe file. In case of Linux kernel those
+variables are named `KBRANCH` and `SRCREV_machine`
+
+Below is list of recipe files for main TB components
+
+```text
+intel-sinit-acm - recipes-support/intel-sinit-acm/intel-sinit-acm_630744.bb
+skl - recipes-support/skl/skl_git.bb
+linux-tb - recipes-kernel/linux/linux-tb_6.6.bb
+grub & grub-efi (common parts) - recipes-bsp/grub/grub-tb-common.inc
+grub - recipes-bsp/grub/grub_%.bbappend
+grub-efi - recipes-bsp/grub/grub-efi_%.bbappend
+```
+
+### Building modified source
+
+In order to make and test changes to source code you first need to enter kas
+shell
+
+```shell
+kas-container shell meta-trenchboot/kas-generic-tb.yml
+```
+
+After that use `devtool modify <recipe>` to fetch source code of recipe you want
+to work on. Bitbake will checkout branch and commit set by `BRANCH` and `SRCREV`
+variables.
+
+```shell
+devtool modify skl
```

Comment:
How long does it take to rebuild a linux this way when adding a simple patch - such as adding a debug print?
It would clean and rebuild everything, right?

If that is the case, this method of development would be totally impractical for TB devs. We may keep it documented, but we should explain it, and be aware that it will likely not be used in the most common scenarios.

Reply:
I just checked with Linux kernel (`linux-tb`) and first build took 5-6 minutes. Second build after adding `printf` to one file took maybe minute or less.

Reply:
Ok, this does not sound too bad, I thought it will be much worse.
I just tested GRUB and SKL. In case of GRUB, I have noticed it always reconfigures the whole repo, even if we add a single character to the print.